assets/dev/js/editor/utils/helpers.js

```diff
@@ -430,6 +430,81 @@ module.exports = {
+	 * performing checks for larger/smaller than, checking if keys/values contain other keys/values, and more.
+	 *
+	 * This method receives a simple Condition (name and value) and converts it into the advanced Conditions format.
+	 * format.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @param conditionName
+	 * @param conditionValue
+	 * @param controlModel - The control being tested.
+	 * @param values - The containing widget's array of control values.
+	 * @param controls - The containing widget's array of control models.
+	 * @returns {{name, value: ({length}|*), operator: (string)}}
+	 */
+	convertConditionToConditions: function( conditionName, conditionValue, controlModel, values, controls ) {
+		// The first step is to isolate the term from the negative operator if exists. For example, a condition format
+		// can look like 'selected_icon[value]!', so we examine this term with a negative connotation.
+		const conditionNameParts = conditionName.match( /([\w-]+(?:\[[\w-]+])?)?(!?)$/i ),
+			conditionRealName = conditionNameParts[ 1 ],
+			isNegativeCondition = !! conditionNameParts[ 2 ];
+
+		const parsedControlName = conditionRealName.match( /([\w-]+)(?:\[([\w-]+)])?/ ),
+			// conditionNameWithoutSubKey example: the condition key 'image[url]' will give the value of 'image'.
+			conditionNameWithoutSubKey = parsedControlName[ 1 ],
+			// conditionSubKey example: the condition key 'image[url]' will give the value of 'url'.
+			conditionSubKey = parsedControlName[ 2 ],
+			// In some cases the control's attributes will be under the 'attributes' property, and in some
+			// cases they will be directly on the model object.
+			controlResponsiveProp = controlModel.attributes?.responsive || controlModel.responsive;
+
+		let conditionNameToCheck = conditionRealName,
+			controlValue;
+
+		// If the conditioning control is responsive, get the appropriate device's value.
+		if ( !! controlResponsiveProp && controls[ conditionNameWithoutSubKey ]?.responsive ) {
+			const queryDevice = controlResponsiveProp.max || controlResponsiveProp.min;
+
+			const deviceSuffix = 'desktop' === queryDevice ? '' : '_' + queryDevice;
+
+			conditionNameToCheck = conditionNameWithoutSubKey + deviceSuffix;
+
+			if ( conditionSubKey ) {
+				conditionNameToCheck += `[${ conditionSubKey }]`;
+			}
+
+			// If the control is not desktop, take the value of the conditioning control of the corresponding device.
+			controlValue = values[ conditionNameWithoutSubKey + deviceSuffix ];
+		} else {
+			controlValue = values[ conditionRealName ];
+		}
+
+		let operator;
+
+		if ( Array.isArray( conditionValue ) && conditionValue.length ) {
+			operator = isNegativeCondition ? '!in' : 'in';
+		} else if ( Array.isArray( controlValue ) && controlValue.length ) {
+			operator = isNegativeCondition ? '!contains' : 'contains';
+		} else if ( isNegativeCondition ) {
+			operator = '!==';
+		}
+
+		return {
+			name: conditionNameToCheck,
+			operator: operator,
```

Comment:
Maybe extract the logic of the operator and the name into 2 different functions
```suggestion
			name: getContionName(...ALL_THE_ARGS_THAT_REQUIRED),
			operator: getOperator(...ALL_THE_ARGS_THAT_REQUIRED),
```

Reply:
For the operator - good idea, implemented.
For the name - It's within an `if` statement with multiple checks, extracting it would mean more code and duplicated code, so I rather avoid it.